### ISSUE TITLE
Verify with iOS 9, 10, and 11

### ISSUE THREAD

**Issue by pasin** [couchbase/couchbase-lite-ios#2100]
From #2097, we have some iOS 9 compatibility issue as we may use APIs that are not available in some of the iOS version. Need to run full unit tests on each version of the iOS (MacOS as well).

**Comment by djpongh** [comment]
@ssmotra, are we targeting iOS9? I thought it was 10+

**Comment by snej** [comment]
Found it: `-Wunguarded-availability`,[ added in Xcode 9.0](https://developer.apple.com/library/content/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html#//apple_ref/doc/uid/TP40001051-CH1-SW2). CBL/iOS already has the build setting `CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE`, but LiteCore doesn't.

I tried adding that setting to LiteCore, and reverting FilePath.cc back to the version with the availability bug, and sure enough I got a compile error:

```
/Couchbase/2.0/vendor/couchbase-lite-core/LiteCore/Support/FilePath.cc:332:51: error: 'kCFURLCanonicalPathKey' is only available on iOS 10.0 or newer [-Werror,-Wunguarded-availability]
        if (!CFURLCopyResourcePropertyForKey(url, kCFURLCanonicalPathKey, &cfCanonPath, outError))
                                                  ^~~~~~~~~~~~~~~~~~~~~~
```

That was the only error, though, so any other iOS 9 issues in LiteCore won't be flushed out this way.

**Comment by djpongh** [comment]
I think #1804 will take care of this. Will close this issue as a dup.
